micro_sam/prompt_generators.py

```diff
@@ -138,12 +138,14 @@ def _sample_negative_points(self, object_mask, bbox_coordinates, coord_list, lab
         # getting the negative points
         # for this we do the opposite and we set the mask to the bounding box - the object mask
         # we need to dilate the object mask before doing this: we use scipy.ndimage.binary_dilation for this
-        # TODO upldate this to use dilation compatible with pytorch (either use dilation or distance trafo from kornia)
-        dilated_object = binary_dilation(object_mask, iterations=self.dilation_strength)
+        dilated_object = morphology.dilation(object_mask[None, None],
+                                             torch.ones(self.dilation_strength, self.dilation_strength))
+        dilated_object = dilated_object.squeeze().numpy()
```

Comment:
I wouldn't cast back to numpy here. That's potentially unnecessary data moving between GPU and CPU.
If you also do
```
background_mask = torch.zeros(object_mask.shape, device=object_mask.device)
```
then everything below should more or less still work. (With some other minor changes, like the `dtype` casts in line 150).

Reply:
Yes, this makes senses (keeping the data in one device and moving ahead with tensors). Took care of this. Thanks for mentioning, Constantin!